include/caffe/util/gpu_memory.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef CAFFE_UTIL_GPU_MEMORY_HPP_
+#define CAFFE_UTIL_GPU_MEMORY_HPP_
+
+#include <vector>
+#include "caffe/common.hpp"
+
+namespace caffe {
+
+class GPUMemoryManager {
+ public:
+  enum PoolMode {
+    NO_POOL,  // Straight CUDA malloc/free (may be expensive)
```

Comment:
Perhaps this should be called `MALLOC`, or something else? I could see other options being added which are not pools...